driver-core/src/test/unit/com/mongodb/internal/connection/ServerDiscoveryAndMonitoringTest.java

```diff
@@ -125,25 +120,30 @@ private ServerDescription getServerDescription(final String serverName) {
     }
 
     private void assertTopologyType(final String topologyType) {
-        if (topologyType.equals("Single")) {
-            assertEquals(SingleServerCluster.class, getCluster().getClass());
-            assertEquals(getClusterType(topologyType, getCluster().getCurrentDescription().getServerDescriptions()),
-                    getCluster().getCurrentDescription().getType());
-        } else if (topologyType.equals("ReplicaSetWithPrimary")) {
-            assertEquals(MultiServerCluster.class, getCluster().getClass());
-            assertEquals(getClusterType(topologyType), getCluster().getCurrentDescription().getType());
-            assertEquals(1, getPrimaries(getCluster().getCurrentDescription()).size());
-        } else if (topologyType.equals("ReplicaSetNoPrimary")) {
-            assertEquals(MultiServerCluster.class, getCluster().getClass());
-            assertEquals(getClusterType(topologyType), getCluster().getCurrentDescription().getType());
-            assertEquals(0, getPrimaries(getCluster().getCurrentDescription()).size());
-        } else if (topologyType.equals("Sharded")) {
-            assertEquals(MultiServerCluster.class, getCluster().getClass());
-            assertEquals(getClusterType(topologyType), getCluster().getCurrentDescription().getType());
-        } else if (topologyType.equals("Unknown")) {
-            assertEquals(getClusterType(topologyType), getCluster().getCurrentDescription().getType());
-        } else {
-            throw new UnsupportedOperationException("No handler for topology type " + topologyType);
+        switch (topologyType) {
```

Comment:
Does this change behavior?  It's hard to tell with all the refactoring diffs intermixed.

Reply:
No that was an auto cleanup

Reply:
Looking more closely, the second assertion for type "Single" was removed.  I'm unsure as to why though.

Reply:
Because now Single topology type doesn't necessarily `SingleServerCluster`.  If directConnection = false it may be single or multiple.

Reply:
I see.  Could we instead make the assertion more nuanced?  Perhaps assert

 `(SingleServerCluster OR (MultiServerCluster AND clusterType == Standalone))`

?  I'm not totally sure what will work but I would like to keep the assertion in some form or another, unless you're convinced that it's not needed.

Reply:
Done